src/main/java/sg/dex/starfish/impl/url/RemoteHttpAsset.java

```diff
@@ -27,118 +24,187 @@
-	 *
-	 * @return An input stream allowing consumption of the asset data
-	 * @throws AuthorizationException if requestor does not have access permission
-	 * @throws StorageException if unable to load the Asset
-	 */
-	@Override
-	public InputStream getContentStream() {
-		HttpGet httpget = new HttpGet(uri);
-		CloseableHttpResponse response = null;
-		try {
-			response = HTTP.execute(httpget);
-			StatusLine statusLine = response.getStatusLine();
-			int statusCode = statusLine.getStatusCode();
-			if (statusCode == 404) {
-				throw new RemoteException("Asset ID not found at: " + uri);
-			}
-			if (statusCode == 200) {
-				InputStream inputStream = HTTP.getContent(response);
-				return inputStream;
-			} else {
-				throw new HttpResponseException(statusLine.getStatusCode(), statusLine.getReasonPhrase());
-			}
-		}
-		catch (IOException e) {
-			throw new RemoteException(" Getting Remote Asset content failed: ", e);
-		}
-
-	}
-
-	/**
-	 * Gets RemoteAsset size
-	 *
-	 * @return size of the RemoteAsset
-	 */
-	@Override
-	public long getContentSize() {
-		try {
-			return getContentStream().available();
-		}
-		catch (IOException e) {
-			throw new GenericException(
-					"Exception occurred  for asset id :" + getAssetID() + " while finding getting the Content size :",
-					e);
-		}
-	}
-
-	@Override
-	public DID getAssetDID() {
-		throw new UnsupportedOperationException("Can't get DID for asset of type " + this.getClass());
-	}
+    /**
+     * This method is to build the metadata of the Resource Asset
+     *
+     * @param uri                     uri
+     * @param metaData                metadata associated with the asset.This metadata will be be added in addition to default
+     *                                metadata i.e DATE_CREATED,TYPE,CONTENT_TYPE.If same key,value is provided then the
+     *                                default value will be overridden.
+     * @param isHashOfContentRequired is Hash Of Content Required
+     * @return String buildMetadata
+     */
+    private static String buildMetaData(String uri, String metaData, boolean isHashOfContentRequired) {
+
+        Map<String, Object> ob = new HashMap<>();
+        ob.put(Constant.DATE_CREATED, Instant.now().toString());
+        ob.put(Constant.TYPE, Constant.DATA_SET);
+        ob.put(Constant.CONTENT_TYPE, "application/octet-stream");
+
+
+        if (metaData != null) {
+
+            for (Map.Entry<String, Object> me : JSON.toMap(metaData).entrySet()) {
+                ob.put(me.getKey(), me.getValue());
+            }
+        }
+
+        if (isHashOfContentRequired) {
+            ob.put(Constant.CONTENT_HASH, getHashContent(getUri(uri)));
+        }
+        return JSON.toString(ob);
+    }
+
+    /**
+     * This method is used to calculate the hash of the content by using keccak256 hashing algorithm.
+     *
+     * @param uri URI
+     * @return the content of hash as string
+     */
+    private static String getHashContent(URI uri) {
```

Comment:
Don't do this. Have a generic Asset.computeContentHash() which uses Asset.getContentStream(). Then you don't need to write separate implementations for each subclass of Asset!